content_sync/pipelines/definitions/concourse/site_pipeline.py

```diff
@@ -0,0 +1,820 @@
+CLEAR_CDN_CACHE_IDENTIFIER = Identifier("clear-cdn-cache").root
+
+
+class SitePipelineDefinitionConfig:
+    """
+    A class with configuration properties for building a site pipeline
+
+    Args:
+        site(Website): The Website object to build the pipeline for
+        pipeline_name(str): The pipeline name to use in Concourse (draft / live)
+        instance_vars(str): The instance vars for the pipeline in a query string format
+        site_content_branch(str): The branch to use in the site content repo (preview / release)
+        static_api_url(str): The base URL for fetching JSON files from the static API (https://ocw.mit.edu/)
+        storage_bucket_name(str): The bucket name ocw-studio stores resources in (ol-ocw-studio-app)
+        artifacts_bucket(str): The versioned bucket where the webpack manifest is stored (ol-eng-artifacts)
+        web_bucket(str): The online bucket to publish output to (ocw-content-draft)
+        offline_bucket(str): The offline bucket to publish output to (ocw-content-offline-draft)
+        resource_base_url(str): A base URL to override fetching of resources from
+        ocw_studio_url(str): The URL to the instance of ocw-studio the pipeline should call home to
+        ocw_hugo_themes_branch(str): The branch of ocw-hugo-themes to use
+        ocw_hugo_projects_branch(str): The branch of ocw-hugo-projects to use
+        hugo_override_args(str): Arguments to override in the hugo command
+    """
+
+    def __init__(
+        self,
+        site: Website,
+        pipeline_name: str,
+        instance_vars: str,
+        site_content_branch: str,
+        static_api_url: str,
+        storage_bucket_name: str,
+        artifacts_bucket: str,
+        web_bucket: str,
+        offline_bucket: str,
+        resource_base_url: str,
+        ocw_studio_url: str,
+        ocw_hugo_themes_branch: str,
+        ocw_hugo_projects_branch: str,
+        hugo_override_args: Optional[str] = "",
+    ):
+        self.site = site
+        self.pipeline_name = pipeline_name
+        self.is_root_website = site.name == settings.ROOT_WEBSITE_NAME
+        if self.is_root_website:
+            self.base_url = ""
+            self.static_resources_subdirectory = f"/{site.get_url_path()}/"
+            self.delete_flag = ""
+        else:
+            self.base_url = site.get_url_path()
+            self.static_resources_subdirectory = "/"
+            self.delete_flag = " --delete"
+        self.site_content_branch = site_content_branch
+        self.pipeline_name = pipeline_name
+        self.static_api_url = static_api_url
+        self.storage_bucket_name = storage_bucket_name
+        self.artifacts_bucket = artifacts_bucket
+        self.web_bucket = web_bucket
+        self.offline_bucket = offline_bucket
+        self.resource_base_url = resource_base_url
+        if (
+            self.site_content_branch == "preview"
+            or settings.OCW_STUDIO_ENVIRONMENT not in PRODUCTION_NAMES
```

Comment:
Should be `settings.ENVIRONMENT` instead.  This will error if `site_content_branch == "release"`

From settings.py:
```python
ENVIRONMENT = get_string(
    name="OCW_STUDIO_ENVIRONMENT",
    default="dev",
    description="The execution environment that the app is in (e.g. dev, staging, prod)",
    required=True,
)
```

Reply:
Good catch, although this should be `settings.ENV_NAME` as seen here: https://github.com/mitodl/ocw-studio/blob/9ecb6e0e65db3c8ead578519bead8af0986a4fb2/content_sync/pipelines/concourse.py#L437